### PR TITLE
Add register page

### PR DESCRIPTION
**Pull request by Porquepix**
- Add register page
- Add 404 page
- Add Secure pages

### CODE REVIEW

**Comment by Porquepix** [comment]
This PR is still in progess please do not merge it :D

**Comment by Porquepix** [comment]
You can now review this PR but please do not merge it until the back is ready.

File: src/components/Navbar/Navbar.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.right.menu .item .button:nth-child(1) {
+    margin-right: 0.5em !important;
```
Comment (QuentinBouygues) [inline src/components/Navbar/Navbar.css]: Why do you need to use `!important` ?

File: src/components/RegisterForm/RegisterForm.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                                />
+                                <FormInput
+                                    fluid={true}
+                                    icon="user"
+                                    iconPosition="left"
+                                    placeholder="Username"
+                                    value={this.state.usernameInput}
+                                    onChange={this.handleUsernameInput}
+                                />                                
+                                <FormInput
```
Comment (QuentinBouygues) [inline src/components/RegisterForm/RegisterForm.tsx]: Sould we add a confirmation field for the password ?

**Pull request merged**